src/components/media-views.js

```diff
@@ -421,6 +427,23 @@ AFRAME.registerComponent("media-video", {
     this.changeVolumeBy(-0.1);
   },
 
+  async snap() {
+    if (this.isSnapping) return;
+    this.isSnapping = true;
+    this.el.sceneEl.systems["hubs-systems"].soundEffectsSystem.playSoundOneShot(SOUND_CAMERA_TOOL_TOOK_SNAPSHOT);
+
+    const canvas = document.createElement("canvas");
+    canvas.width = this.video.videoWidth;
+    canvas.height = this.video.videoHeight;
+    canvas.getContext("2d").drawImage(this.video, 0, 0, canvas.width, canvas.height);
+    const blob = await new Promise(resolve => canvas.toBlob(resolve));
+    const file = new File([blob], "snap.png", { type: "image/png" });
+
+    this.localSnapCount++;
+    const { entity } = spawnMediaAround(this.el, file, this.localSnapCount);
+    entity.addEventListener("image-loaded", () => (this.isSnapping = false), { once: true });
```

Comment:
Write `this.onImageLoaded = () => { this.isSnapping = false; }; ` in `init`, then replace this line with
`entity.addEventListener("image-loaded", this.onImageLoaded, { once: true });` 
to avoid creating garbage.
If you want to, you can even write
`const ONCE_TRUE = { once: true }` at the top of the file and then use that when adding an event listener to avoid creating an object. 

Not a big deal because this snapping happens rarely, but I've started to avoid allocations whenever possible unless it adds significant complexity.